docs/migration_route.md

```diff
@@ -0,0 +1,41 @@
+# Migration Route
+
+Migrations are available via the internal route `/internal/migrations/` with the following payload:
+```js
+{
+    "cmd": "migrate" | "finalize" | "reset" | "clear-collectionfield-tables" | "stats" | "progress",
+    "verbose": bool
+}
+```
+
+All commands except `progress` are directly translated the the respective method calls to the datastore module as well as the `verbose` flag. While a migration is running (either via `migrate` or `finalize`), no other commands are permitted except `progress`. The `progress` command is an additional command only available in the backend which reports the progress of a long-running migration (which is executed in a thread).
+
+The output of all commands is the following (for a successful request):
+```js
+enum MigrationProgressState {
+    NO_MIGRATION_RUNNING = 0
+    MIGRATION_RUNNING = 1
+    MIGRATION_FINISHED = 2
+}
+
+{
+    "success": true,
+    "status"?: MigrationProgressState,
+    "output"?: str,
+    "exception"?: str,
+}
+```
+The output differs slightly, depending on the command:
+- `migrate`/`finalize`/`progress`: `status` can only be 0 if `progress` is called while no migration is running, otherwise it will always be greater than 0. If `status` is 1, `output` contains the output until now of either the command issued in this request or, in the case of a `progress` command, of the threaded migration. If the migration is finished (`status=2`), the full output is returned and stored until a new migration is started, meaning repeated `progress` requests after this point will always return the same result with the full output.
```

Comment:
Is it true that if migrations are required and I send the migrate or finalize command (not progress) the handler returns immediately and status is always 1? It can never be 2 in this case, right? What if no migrations are required? Is status then 0 or 2? If it is 2 I think this is missleading because there aren't any migrations that were recently finished.

I propose the following changes:

```js
enum MigrationProgressState {
    NO_MIGRATIONS_REQUIRED = "no_migrations_required"
    MIGRATIONS_REQUIRED = "migrations_required"
    MIGRATIONS_RUNNING = "migrations_running"
    PREPARED_MIGRATIONS_PENDING = "prepared_migrations_pending"
}
```

Of course the flags can be less verbose but this is what they should tell me.

Migrate, finalize progress and stats should always return a status. Only for reset and clear-collectionfied-tables the status can be omitted.

I also strongly recommend using strings and not integers. This is more verbose for the user of this API (that's me in the manage tool). If you insist in integers, I can live with it but then please do not use 0. The 0 should be the same as a not given property. This would make many things a lot easier for me in Golang code.

Reply:
As discussed: Status also for reset.